src/Middleware/RateLimiting/src/RateLimitingMiddleware.cs

```diff
@@ -60,6 +60,13 @@ public RateLimitingMiddleware(RequestDelegate next, ILogger<RateLimitingMiddlewa
     /// <returns>A <see cref="Task"/> that completes when the request leaves.</returns>
     public async Task Invoke(HttpContext context)
     {
+        var endpoint = context.GetEndpoint();
+        // If this endpoint has a DisableRateLimitingAttribute, don't apply any rate limits.
+        if (endpoint?.Metadata.GetMetadata<DisableRateLimitingAttribute>() is not null)
+        {
+            await _next(context);
```

Comment:
nit: Separate the logic below into a different method so we can remove await and async from the main middleware `Invoke` path.

Reply:
Wonder if we should also try to fast-path the case where no limiter will be applied, e.g. when `_globalLimiter` is null and there is no `EnableRateLimitingAttribute` on the endpoint.